File: cs50x/Psets/pset6/readability/readability.py
# Import get_string from cs50 library
from cs50 import get_string

# Ask the user for text
text = get_string('Text: ')

# Counter
counter = 0

# Count how many letters are in the text
for i in text:
    if i.isalpha():
        counter += 1
l = counter
counter = 0

# Count how many words are in the text
for i in text:
    if i.isspace():
        counter += 1
w = counter
if text[len(text) - 1] == '!' or text[len(text) - 1] == '.' or text[len(text) - 1] == '?':
    w += 1
counter = 0

# Count how many senteences are in the text
for i in range(len(text)):
    if text[i] == '!' or text[i] == '.' or text[i] == '?':
        counter += 1
s = counter

# Average letters
r = l * 100
R = r / w
L = R

# Average sentences
r = s * 100
R = r / w
S = R

# Coleman-Liau index
index = 0.0588 * L - 0.296 * S - 15.8
index = round(index)

# Print the grade
if index > 17:
    print('Grade 16+')
elif index < 0:
    print('Before Grade 1')
else:
    print('Grade ' + str(index))

File: cs50x/Psets/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
float rule_three(int l, int w);
float rule_three2(int s, int w);


int main(void)
{
    // Ask the user for text
    string text = get_string("Text: ");
    
    // Count letters, words and senteences
    int l = count_letters(text);
    int w = count_words(text);
    int s = count_sentences(text);
    
    // Average number of letters
    float L = rule_three(l, w);
    // Average number of sentences
    float S = rule_three2(s, w);
    
    //  Coleman-Liau index
    float index = 0.0588 * L - 0.296 * S - 15.8;
    // round de index
    int grade = round(index);
    
    // Print the grade
    if (grade > 17)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 0)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

// Function for counting letters
int count_letters(string text)
{
    int counter = 0;
    for (int a = 0; a < strlen(text); a++)
    {
        if (isalpha(text[a]))
        {
            counter++;
        }
    }
    return counter;
}

// Function for counting words
int count_words(string text)
{
    int word = 0;
    for (int b = 0; b < strlen(text); b++)
    {
        if (isspace(text[b]))
        {
            word++;
        }
    }
    word++;
    return word;
}

// Function for counting sentences
int count_sentences(string text)
{
    int sentences = 0;
    for (int c = 0; c < strlen(text); c++)
    {
        if (text[c] == 46 || text[c] == 63 || text[c] == 33)
        {
            sentences++;
        }
    }
    return sentences;
}

// Function for the average letters
float rule_three(int l, int w)
{
    float r = l * 100;
    float R = r / w;
    return R;
}

// Function for the average sentences
float rule_three2(int s, int w)
{
    float r = s * 100;
    float R = r / w;
    return R;
}